### ISSUE TITLE
[Unity 2020.3] ZString's StringBuilder slower than System.StringBuilder?

### ISSUE THREAD

**Issue by Lf3T-Hn4D** [Cysharp/ZString#47]
When reviewing this wonderful StringBuilder replacement, I noticed the performance for Append() with string type is much worse compared to System.Text.StringBuilder.

![image](https://user-images.githubusercontent.com/7812588/114486399-eacd0e80-9c61-11eb-9e39-ead21b4ba6b2.png)

The test code I'm using is:
```c#
int COUNT = 1000;
List<string> strings = new List<string>();
for (int i = 0; i < 100; i++)
	strings.Add("123456789");

Profiler.BeginSample("Append/SharedStringBuilderScope()");
{
	System.Text.StringBuilder sb = new System.Text.StringBuilder();
	for (int i = 0; i < COUNT; i++)
	{
		for (int j = 0; j < strings.Count; j++)
		{
			sb.Append(strings[j]);
		}

		sb.ToString();
		sb.Clear();
	}
}
Profiler.EndSample();

Profiler.BeginSample("Append/ZString.StringBuilder()");
for (int i = 0; i < COUNT; i++)
{
	using (var sb = ZString.CreateStringBuilder())
	{
		for (int j = 0; j < strings.Count; j++)
		{
			sb.Append(strings[j]);
		}

		sb.ToString();
	}
}
Profiler.EndSample();
```

Is this a known issue? Based on deep profiling in Unity, I noticed that the bottleneck was caused by explicit IntPtr casting. I'm not sure if this is just a bogus timing due to deep profiling, but making IL2CPP builds also shows it to be performing poorly vs System.Text.StringBuilder

**Comment by neuecc** [comment]
at first, StringBuilder is reused(`sb.Clear`) but ZString is not.

modified
```
	for (int i = 0; i < COUNT; i++)
	{
         	System.Text.StringBuilder sb = new System.Text.StringBuilder();
		for (int j = 0; j < strings.Count; j++)
```

shows win for allocate but yes, perf is still slow.

This is caused by SlowSpan, ZString's internal architecture is Span based.
for .NET Core uses FastSpan in Span but for Unity uses SlowSpan, it casused IntPtr casting, etc.

However, in the very common Append(string) scenario, casting to Span is likely to be avoided.
The performance test with the modified code is shown below.

![image](https://user-images.githubusercontent.com/46207/114491244-45aa3a00-9c51-11eb-9b31-1c4d012e367b.png)

Thank you, I would like to release this code as soon as possible.

**Comment by Lf3T-Hn4D** [comment]
That looks better! And yes, I was reusing the StringBuilder because in our projects, we have a shared string builder that we reuse. I was doing a review to compare what benefit ZString will provide if we switch over. :-)

**Comment by neuecc** [comment]
I've released `2.4.1`, it includes this optimization. thanks!

**Comment by Lf3T-Hn4D** [comment]
That was quick. Thanks!
